src/lightning_utilities/core/enums.py

```diff
@@ -19,46 +18,29 @@ class StrEnum(str, Enum):
     True
     >>> MySE.from_str("t-2", source="value") == MySE.t2
     True
+    >>> MySE.from_str("t-2", source="value")
+    <MySE.t2: 'T-2'>
+    >>> MySE.from_str("t-3", source="any")
+    Traceback (most recent call last):
+      ...
+    ValueError: Invalid match: expected one of ['t1', 't2', 'T-1', 'T-2'], but got t-3.
     """
 
     @classmethod
-    def from_str(
-        cls, value: str, source: Literal["key", "value", "any"] = "key", strict: bool = False
-    ) -> Optional["StrEnum"]:
-        """Create StrEnum from a sting matching the key or value.
+    def from_str(cls, value: str, source: Literal["key", "value", "any"] = "key") -> "StrEnum":
```

Comment:
this should always return an instance of this enum, to avoid None checks. if one wants to allow failures, the ValuError can be matched

```
try:
    x = from_str():
except ValueError:
    x = None

Reply:
Why this should always return an instance of this enum even you ask nonsense?

Reply:
Because otherwise mypy will force us to `assert x is not None` everytime after. I added an utility that catches the valuerror for those who want this behaviour

Reply:
so what are you saying, return Enum or raise error, not return Enum always?